src/frontend/src/CustomNodes/GenericNode/components/NodeInputField/index.tsx

```diff
@@ -0,0 +1,175 @@
+import useHandleNodeClass from "@/CustomNodes/hooks/use-handle-node-class";
+import { ParameterRenderComponent } from "@/components/parameterRenderComponent";
+import { usePostTemplateValue } from "@/controllers/API/queries/nodes/use-post-template-value";
+import { ReactNode, useEffect, useRef, useState } from "react";
+import { default as IconComponent } from "../../../../components/genericIconComponent";
+import ShadTooltip from "../../../../components/shadTooltipComponent";
+import { LANGFLOW_SUPPORTED_TYPES } from "../../../../constants/constants";
+import useFlowStore from "../../../../stores/flowStore";
+import { useTypesStore } from "../../../../stores/typesStore";
+import {
+  NodeInputFieldComponentType,
+  ParameterComponentType,
+} from "../../../../types/components";
+import { scapedJSONStringfy } from "../../../../utils/reactflowUtils";
+import useFetchDataOnMount from "../../../hooks/use-fetch-data-on-mount";
+import useHandleOnNewValue from "../../../hooks/use-handle-new-value";
+import HandleRenderComponent from "../handleRenderComponent";
+
+export default function NodeInputField({
+  id,
+  data,
+  tooltipTitle,
+  title,
+  colors,
+  type,
+  name = "",
+  required = false,
+  optionalHandle = null,
+  info = "",
+  proxy,
+  showNode,
+}: NodeInputFieldComponentType): JSX.Element {
+  const ref = useRef<HTMLDivElement>(null);
+  const infoHtml = useRef<HTMLDivElement & ReactNode>(null);
+  const nodes = useFlowStore((state) => state.nodes);
+  const edges = useFlowStore((state) => state.edges);
+  const myData = useTypesStore((state) => state.data);
+  const postTemplateValue = usePostTemplateValue({
+    node: data.node!,
+    nodeId: data.id,
+    parameterId: name,
+  });
+  const setFilterEdge = useFlowStore((state) => state.setFilterEdge);
+  const { handleNodeClass } = useHandleNodeClass(data.id);
+
+  let disabled =
+    edges.some(
+      (edge) =>
+        edge.targetHandle === scapedJSONStringfy(proxy ? { ...id, proxy } : id),
+    ) ?? false;
+
+  const { handleOnNewValue } = useHandleOnNewValue({
+    node: data.node!,
+    nodeId: data.id,
+    name,
+  });
+
+  useFetchDataOnMount(data.node!, handleNodeClass, name, postTemplateValue);
+
+  useEffect(() => {
+    // @ts-ignore
+    infoHtml.current = (
+      <div className="h-full w-full break-words">
+        {info.split("\n").map((line, index) => (
+          <p key={index} className="block">
+            {line}
+          </p>
+        ))}
+      </div>
+    );
+  }, [info]);
+
+  useEffect(() => {
+    if (optionalHandle && optionalHandle.length === 0) {
+      optionalHandle = null;
+    }
+  }, [optionalHandle]);
+
+  const displayHandle =
+    !LANGFLOW_SUPPORTED_TYPES.has(type ?? "") || optionalHandle;
+
+  return !showNode ? (
```

Comment:
maybe we can refactor this code so that we dont need to instantiate the handleRenderComponent twice

Reply:
that's a good idea, let's improve each component in different pull requests